packages/tldraw/src/lib/shapes/image/ImageShapeUtil.tsx

```diff
@@ -63,9 +63,25 @@ export class ImageShapeUtil extends BaseBoxShapeUtil<TLImageShape> {
 		const prefersReducedMotion = usePrefersReducedMotion()
 		const [staticFrameSrc, setStaticFrameSrc] = useState('')
 		const [loadedSrc, setLoadedSrc] = useState('')
+		const [temporarySrc, setTemporarySrc] = useState('')
 		const isSelected = shape.id === this.editor.getOnlySelectedShapeId()
 		const { asset, url } = useAsset(shape.id, shape.props.assetId, shape.props.w)
 
+		useEffect(() => {
```

Comment:
ach! it's a good idea but it's tricky because of the disposal piece (discussed in the thread below). i need to be able to `clearTemporaryAssetPreview` once the image is ready (`loadedSrc` is not null)